### PR TITLE
Update Eclipse Paho to version 1.4.1

### PR DESCRIPTION
**Pull request by antonia-avramova**
[#7] Update Eclipse Paho to version 1.4.1

Signed-off-by: Antonia Avramova <antonia.avramova@bosch.io>

### CODE REVIEW

**Comment by e-grigorov** [comment]
@antonia-avramova The Eclipse Kanto project is currently restructuring its license scheme - from Eclipse Public License 2.0 only to dual licensing under Eclipse Public License 2.0 **and** Apache License 2.0 (you can read more about it [here](https://projects.eclipse.org/projects/iot.kanto/reviews/restructuring-review)).

This contribution will be merged into the project with this new licensing scheme so its important for the project and the community to have your agreement on proceeding with it given these new circumstances :blush: 

Please, provide your feedback as a comment here that you, as an author of this contribution, understand and agree with the change explained above. Thank you!

File: go.sum [hunk 1]
```diff
@@ -36,6 +36,8 @@ github.com/eclipse/ditto-clients-golang v0.0.0-20220225085802-cf3b306280d3 h1:bf
 github.com/eclipse/ditto-clients-golang v0.0.0-20220225085802-cf3b306280d3/go.mod h1:ey7YwfHSQJsinGkGbgeEgqZA7qJnoB0YiFVTFEY50Jg=
 github.com/eclipse/paho.mqtt.golang v1.3.5 h1:sWtmgNxYM9P2sP+xEItMozsR3w0cqZFlqnNN1bdl41Y=
```
Comment (e-grigorov) [inline go.sum:37]: Could you please run: `go mod tidy`, it'll remove this line?

File: NOTICE.md [hunk 2]
```diff
@@ -153,6 +153,12 @@ golang.org/x/time (v0.0.0-20190308202827-9d24e82272b4)
 * Project: https://github.com/golang/time
 * Source:  https://github.com/golang/time/tree/9d24e82272b4f38b78bc8cff74fa936d31ccd8ef
 
+golang.org/x/sync (v0.0.0-20210220032951-036812b2e83c)
```
Comment (e-grigorov) [inline NOTICE.md:156]: To keep the readability, I would suggest to put it between net and sys.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

**Comment by antonia-avramova** [comment]
> @antonia-avramova The Eclipse Kanto project is currently restructuring its license scheme - from Eclipse Public License 2.0 only to dual licensing under Eclipse Public License 2.0 **and** Apache License 2.0 (you can read more about it [here](https://projects.eclipse.org/projects/iot.kanto/reviews/restructuring-review)).
> 
> This contribution will be merged into the project with this new licensing scheme so its important for the project and the community to have your agreement on proceeding with it given these new circumstances 😊
> 
> Please, provide your feedback as a comment here that you, as an author of this contribution, understand and agree with the change explained above. Thank you!

I understand and agree with the license change explained above.

**Review by e-grigorov — APPROVED** [review]

**Pull request merged**